# HG changeset patch
# User Vsevolod Volkov <vvv@mutt.org.ua>
# Date 1454260630 0
#      Sun Jan 31 17:17:10 2016 +0000
# Node ID 76ce4f77201c511b77d6cd81b651de3606812011
# Parent  8c85bbb5d7ccf61460659e4a30a99f0a19b87dcc
[PATCH] feature: initials
Expando for author's initials

https://mutt.org.ua/download/mutt-1.9.0/patch-1.9.0.vvv.initials.gz

---
 hdrline.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff -r 8c85bbb5d7cc -r 76ce4f77201c PATCHES
--- a/PATCHES	Tue Jan 26 23:00:09 2016 +0000
+++ b/PATCHES	Sun Jan 31 17:17:10 2016 +0000
@@ -1,3 +1,4 @@
+patch-initials-vvv
 patch-ifdef-neomutt
 patch-allow-slow-IO-interrupt-neomutt
 patch-browser-sort-additions-neomutt
diff -r 8c85bbb5d7cc -r 76ce4f77201c hdrline.c
--- a/hdrline.c	Tue Jan 26 23:00:09 2016 +0000
+++ b/hdrline.c	Sun Jan 31 17:17:10 2016 +0000
@@ -228,6 +228,7 @@ static char *apply_subject_mods (ENVELOP
  * %f = entire from line
  * %F = like %n, unless from self
  * %i = message-id
+ * %I = initials of author
  * %l = number of lines in the message
  * %L = like %F, except `lists' are displayed first
  * %m = number of messages in the mailbox
@@ -478,6 +479,31 @@ hdr_format_str (char *dest,
       mutt_format_s (dest, destlen, prefix, hdr->env->message_id ? hdr->env->message_id : "<no.id>");
       break;
 
+    case 'I':
+      {
+	int iflag = FALSE;
+	int j = 0;
+
+	for (i = 0; hdr->env->from && hdr->env->from->personal &&
+		    hdr->env->from->personal[i] && j < SHORT_STRING - 1; i++)
+	{
+	  if (isalpha ((int)hdr->env->from->personal[i]))
+	  {
+	    if (!iflag)
+	    {
+	      buf2[j++] = hdr->env->from->personal[i];
+	      iflag = TRUE;
+	    }
+	  }
+	  else
+	    iflag = FALSE;
+	}
+
+	buf2[j] = '\0';
+      }
+      mutt_format_s (dest, destlen, prefix, buf2);
+      break;
+
     case 'l':
       if (!optional)
       {
